### PR TITLE
feat: Add gRPC listener and change configs to use grpc and ws

### PR DESCRIPTION
**Pull request by red-0ne**
Add gRPC urls to `AppGateServer` and `RelayMiner` configurations by:
* Replacing `query_node_url` and `network_node_url` with `query_node_grpc_url` and `tx_node_grpc_url` respectively
* Adding `query_node_rpc_url` for subscription queries

_Note: This PR should be merged right before `Poktroll`'s PR [#257](https://github.com/pokt-network/poktroll/pull/257/files#diff-c2ee8653e1d6b85f0aadf87cd438a9250806c052877248442be4d434cbc52425) since `AppGateServer` and `RelayMiner` config changes will break e2e tests that do not have the adequate changes applied_

### CODE REVIEW

File: charts/appgate-server/values.yaml
Comment (Olshansk) [inline charts/appgate-server/values.yaml:28]: We have `query_node_grpc_url` and `pocket_node_websocket_url`. 

I feel like `query_node_grpc_url` and `query_node_websocket_url` would be more appropriate given that the comments are saying that they're both for reading / listening for on-chain data. You can "query" data via poll or listening based mechanisms.

I realize the comments may not be reflected in how it's actually being used so wanted to discuss the naming before we move forward.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: charts/appgate-server/values.yaml
Comment (red-0ne) [reply charts/appgate-server/values.yaml:28]: So we'll have `query_node_grpc_url`, `query_node_websocket_url` and `network_node_grpc_url` right?

If that's the case, I'm OK with it and will reflect these changes here and to Pocktroll corresponding PR

Comment (Olshansk) [reply charts/appgate-server/values.yaml:28]: Rough Notes from call

---

```bash
# Pocket node URL that exposes CometBFT JSON-RPC API.
# This can be used by the cosmos client SDK, event subscriptions, etc...
`query_node_rpc_url`

# Pocket node URL that exposes the Cosmos gRPC service.
`tx_node_grpc_url`

# Pocket node URL that exposes the Cosmos gRPC service, dedicated to querying purposes.
# If unspecified, defaults to `tx_node_grpc_url`.
`query_node_grpc_url`
```

--- 

# Context:
Cosmos Client:
- QueryClient
	- QueryNodeURL
- TxClient
    - TxNodeURL
   
## Tentative changes 1 
Changes if we go with path:    
1. Rename `NetworkNodeUrl` to `TxNodeUrl` so that naming is consistent
2. Update comments to explain that `QueryUrl` is for `CosmosQueryClient` and `TxUrl` is for `CosmosTxClient`

Questions:
- Why are we even doing this?
- What are we trying to achieve?
- Should we even switch to gRPC?
- Are we even switching to something more complex?

What is the problem we're solving?
- `Goal`: Import the SDK without requiring an HTTP server
   - Path to achieve to goal: import SDK w/o running an external binary
- `Problem`: SDK is dependant on cosmos context
- `Solution`: Refactoring the SDK to build a gRPC client instead of a cosmos client

- `Original problem`:
   - Prevent the SDK from being dependent on the cosmos SDK content
   - Start `AppGateServer` and `RelayMiner` independent of `controlled binary`

Some motivation:
- Enable SDK to use gRPC
- Continue using the regular RPC client


Two endpoints:
1. gRPC endpoint
   1.1 Query endpoint (querying or polling)
   1.2 Tx endpoint (send or push)
2. WebSocket endpoint (event subscriptions - listening)

Requirements:
- `1` & `2` are needed at a minimum
- `1.1` can be the same as `1.2` but not a hard requirement (default to the same value if unspecified)

**Review by Olshansk — APPROVED** [review]
I REALLY ❤️ this change after our discussion.

Approving but please wait for an approval/review from @okdas before merging because I don't know if it might break something in the interim.

**Comment by Olshansk** [comment]
@red-0ne Don't forget to update ticket metadata in the PR even if it's a different repo!

**Comment by red-0ne** [comment]
> Approving but please wait for an approval/review from @okdas before merging because I don't know if it might break something in the interim.

:100:  It will break poktroll e2e tests if `poktroll` dos not have changes from https://github.com/pokt-network/poktroll/pull/257. We need to wait for the later to be approved and then merge them both.

**Review by okdas — APPROVED** [review]
After we merge this, we can bump the version of helm charts, so the new version of charts will be published to the Helm repo.

**Pull request merged**
